common/persistence/sql/sqlExecutionStore.go

```diff
@@ -255,6 +254,103 @@ func (m *sqlExecutionStore) GetWorkflowExecution(
 			Message: fmt.Sprintf("GetWorkflowExecution return more than one results."),
 		}
 	}
+	return executions, nil
+}
+
+func (m *sqlExecutionStore) GetWorkflowExecution(
+	ctx context.Context,
+	request *p.InternalGetWorkflowExecutionRequest,
+) (*p.InternalGetWorkflowExecutionResponse, error) {
+
+	domainID := serialization.MustParseUUID(request.DomainID)
+	runID := serialization.MustParseUUID(request.Execution.RunID)
+	wfID := request.Execution.WorkflowID
+
+	// get executions
+	executionsChan := make(chan []sqlplugin.ExecutionsRow)
+	executionsErrChan := make(chan error)
+	go func() {
+		executions, err := m.getExecutions(ctx, request, domainID, wfID, runID)
+		executionsChan <- executions
+		executionsErrChan <- err
+	}()
```

Comment:
with an errgroup this would probably be:
```go
var executions []sqlplugin.ExecutionsRow
g.Go(func() error {
  var err error
  executions, err = m.getEsxecutions(ctx, request, domainID, wfID, runID)
  return err
})
...
err := g.Wait()
if err != nil {
  return nil, err
}
// executions is populated and not racy, use it
```